mvpa2/tests/test_generators.py

```diff
@@ -396,3 +397,46 @@ def is_sorted(x):
     permutation = AttributePermutator(attr='targets',
                                       strategy='chunks')
     assert_raises(ValueError, permutation, ds)
+
+
+def test_factorialpartitioner():
+    # Test against sifter and chainmap implemented in test_usecases
+    # -- code below copied from test_usecases --
+    # Let's simulate the beast -- 6 categories total groupped into 3
+    # super-ordinate, and actually without any 'superordinate' effect
+    # since subordinate categories independent
+    ds = normal_feature_dataset(nlabels=6,
+                                snr=100,   # pure signal! ;)
+                                perlabel=30,
+                                nfeatures=6,
+                                nonbogus_features=range(6),
+                                nchunks=5)
+    ds.sa['subord'] = ds.sa.targets.copy()
+    ds.sa['superord'] = ['super%d' % (int(i[1])%3,)
+                         for i in ds.targets]   # 3 superord categories
+    # let's override original targets just to be sure that we aren't relying on them
+    ds.targets[:] = 0
+
+    npart = ChainNode([
+        ## so we split based on superord
+        NFoldPartitioner(len(ds.sa['superord'].unique),
+                         attr='subord'),
+        ## so it should select only those splits where we took 1 from
+        ## each of the superord categories leaving things in balance
+        Sifter([('partitions', 2),
+                ('superord',
+                 { 'uvalues': ds.sa['superord'].unique,
+                   'balanced': True})
+                ]),
+    ], space='partitions')
+
+    # now the new implementation
+    factpart = FactorialPartitioner(
+        NFoldPartitioner(attr='subord'),
+        attr='superord'
+    )
+
+    partitions_npart = [p.sa.partitions for p in npart.generate(ds)]
+    partitions_factpart = [p.sa.partitions for p in factpart.generate(ds)]
+
+    assert_array_equal(np.sort(partitions_npart), np.sort(partitions_factpart))
```

Comment:
I actually wonder... Probably it shouldn't blow if superords aren't balancedd in subords, since it might be a legit case and people might decide to use it with Balancer to mitigate